compiler/rustc_query_system/src/query/job.rs

```diff
@@ -211,14 +211,14 @@ struct QueryLatchInfo<D: DepKind> {
 #[cfg(parallel_compiler)]
 #[derive(Clone)]
 pub(super) struct QueryLatch<D: DepKind> {
-    info: Lrc<Mutex<QueryLatchInfo<D>>>,
+    info: Lrc<Lock<QueryLatchInfo<D>>>,
```

Comment:
I don't think we need to use `Lock` here, since this code is already under `#[cfg(parallel_compiler)]`. Using `Mutex` makes it clearer what this code does

Reply:
Same down there : ) And thanks for pushing this!

Reply:
This is exactly the reason: Other things from `rustc_data_structures::sync` imported and used only in `parallel_compiler` (from 20-30 lines), except for `Mutex`; which for someone who don't know that `Lock` in that case is `Mutex`, so this clean up things, to prevent it be `Mutex` and `Lock` at the same time.

Reply:
I think we could stop using the sync module in cfg-parallel-rustc code entirely and just use the actual data structures.

That would also solve the ambiguity

Reply:
This is a matter. I agree with oli. We should not use data structures in `sync` module when the code will only be executed under multi-threading. I think we can clean up the `Lrc` and `Lock` around these code.